### PR TITLE
Manage buyer organisations and users search bar

### PR DESCRIPTION
**Pull request by magnusjohn-bjss**

### CODE REVIEW

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/OrganisationsControllerTests.cs [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+using Moq;
+using NHSD.GPIT.BuyingCatalogue.EntityFramework.Organisations.Models;
+using NHSD.GPIT.BuyingCatalogue.ServiceContracts.Organisations;
+using NHSD.GPIT.BuyingCatalogue.Test.Framework.AutoFixtureCustomisations;
 using NHSD.GPIT.BuyingCatalogue.WebApp.Areas.Admin.Controllers;
+using NHSD.GPIT.BuyingCatalogue.WebApp.Areas.Admin.Models.OrganisationModels;
+using NHSD.GPIT.BuyingCatalogue.WebApp.Models.Autocomplete;
 using Xunit;
 
 namespace NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests.Areas.Admin.Controllers
```
[the remark this answers is not in the document]
Comment (magnusjohn-bjss) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/OrganisationsControllerTests.cs:20]: test case 'Get_SearchResults_ValidSearchTerm_NoMatches_ReturnsExpectedResult' added

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/OrganisationsController.cs [hunk 2]
```diff
@@ -32,6 +36,36 @@ public sealed class OrganisationsController : Controller
...
+                })
+                .ToList();
+
+            return View(new IndexModel(search, organisationModel));
+        }
+
+        [HttpGet("search-results")]
+        public async Task<IActionResult> SearchResults([FromQuery] string search)
+        {
+            var results = await GetFilteredOrganisations(search);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/OrganisationsController.cs:59]: one again, why not use OrganisationsService.GetOrganisationsFiltered?

**Review by leerogers-bjss — DISMISSED** [review]

**Review by bjss-colin — DISMISSED** [review]
Dismissing my stale review, looks good from my perspective. Pending subsequent reviews from Lee & Oke

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/OrganisationsController.cs (hunk 2 above)
Comment (magnusjohn-bjss) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/OrganisationsController.cs:59]: moved filter logic to service layer

File: src/NHSD.GPIT.BuyingCatalogue.Services/Organisations/OrganisationsService.cs [hunk 3]
```diff
@@ -66,6 +66,15 @@ public async Task<List<Organisation>> GetOrganisationsByInternalIdentifiers(stri
             return await dbContext.Organisations.Where(o => internalIdentifiers.Contains(o.InternalIdentifier)).OrderBy(o => o.Name).ToListAsync();
         }
 
+        public async Task<List<Organisation>> GetOrganisationsBySearchTerm(string searchTerm)
+        {
+            return await dbContext.Organisations
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.Services/Organisations/OrganisationsService.cs:71]: please limit to max of 15 with .Take(15)

**Review by leerogers-bjss — CHANGES_REQUESTED** [review]
just a minor change then will approve

File: src/NHSD.GPIT.BuyingCatalogue.Services/Organisations/OrganisationsService.cs (hunk 3 above)
Comment (magnusjohn-bjss) [reply src/NHSD.GPIT.BuyingCatalogue.Services/Organisations/OrganisationsService.cs:71]: added this at the controller level as it's a ui constraint

**Review by Okechukwu802 — DISMISSED** [review]

**Review by leerogers-bjss — APPROVED** [review]

**Review by Okechukwu802 — APPROVED** [review]

**Pull request merged**